### PR TITLE
Doc(848333): Need to add the detail template exporting sample in UG of all platforms

### PR DESCRIPTION
**Pull request by rajapandiyan-settu**
Added detail template exporting topic in PDF and Excel exporting of grid component.

### CODE REVIEW

File: ej2-javascript/grid/excel-export/exporting-detail-template-grid.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+platform: ej2-javascript
+control: Exporting detail template grid 
+publishingplatform: ##Platform_Name##
+documentation: ugs
+domainurl: ##DomainURL##
+---
+
+# Exporting detail template grid in ##Platform_Name## Grid control
+
+The grid has the option to export the detail template grid to an Excel document. By default, it will use the `Expanded` as hierarchyExportMode. Change the exporting option by using the `ExcelExportProperties.hierarchyExportMode` property. The available options are:
```
Comment (MaithiliyK) [inline ej2-javascript/grid/excel-export/exporting-detail-template-grid.md:14]: provide api link

File: ej2-javascript/grid/excel-export/exporting-detail-template-grid.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+| Expanded | Exports the visible detail rows in the expanded state. |
+| All      | Exports all the detail rows in the expanded state. |
+| None     | Exports all detail rows in a collapsed state. |
+
+Detail rows in the exported Excel can be customized or formatted using the [`exportDetailTemplate`](../../api/grid/#exportdetailtemplate) event. In this event, the detail rows of the Excel document are formatted with respect to their parent row details.
+
+In the following sample, the detail template value has been structured with rows and cells format by specifying the [`columnHeader`](../../api/grid/detailTemplateProperties/#columnheader) and [`rows`](../../api/grid/detailTemplateProperties/#rows) parameters using its [`parentRow`](../../api/grid/exportDetailTemplateEventArgs/#parentrow) details. This allows us to create detail rows in the Excel document. Additionally, apply custom styles to specific cells using the [`style`](../../api/grid/detailTemplateCell/#style) property.
+
+> When using [`rowSpan`](../../api/grid/detailTemplateCell/#rowspan), it's essential to provide the cell's [`index`](../../api/grid/detailTemplateCell/#index) for proper functionality.
```
Comment (MaithiliyK) [inline ej2-javascript/grid/excel-export/exporting-detail-template-grid.md:26]: if you give URL for APi dont give backtick

File: ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+platform: ej2-javascript
+control: Exporting detail template grid 
+publishingplatform: ##Platform_Name##
+documentation: ug
+domainurl: ##DomainURL##
+---
+
+# Exporting detail template grid in ##Platform_Name## Grid control
+
+The grid has the option to export the detail template to a PDF document. By default, the grid will export the master grid with expanded detail rows alone. Change the exporting option by using the `PdfExportProperties.hierarchyExportMode` property. The available options are:
```
Comment (MaithiliyK) [inline ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md:14]: The grid has an option to export the detail template to a PDF document. By default, the grid will export the parent grid with expanded detail rows alone. use this sentence for excel exporting too

File: ej2-javascript/grid/ej2-typescript-grid-toc.html [hunk 4]
```diff
@@ -126,6 +126,7 @@
 <li><a href="/ej2-typescript/grid/pdf-export/pdf-cell-style-customization">PDF Cell Style Customization</a></li>
 <li><a href="/ej2-typescript/grid/pdf-export/adding-header-and-footer">Adding Header and Footer</a></li>
 <li><a href="/ej2-typescript/grid/pdf-export/exporting-hierarchy-grid">Exporting Hierarchy Grid</a></li>
+<li><a href="/ej2-typescript/grid/pdf-export/exporting-detail-template-grid">Exporting Detail Template Grid</a></li>
```
Comment (MaithiliyK) [inline ej2-javascript/grid/ej2-typescript-grid-toc.html:129]: create the structure like this


PDF Exporting with Templates
•	Exporting with column template
•	Exporting with detail template
•	Exporting with caption template

File: ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+| All      | Exports the master grid with all the detail rows. |
+| None     | Exports the master grid alone. |
+
+Detail rows in the exported PDF can be customized or formatted using the [`exportDetailTemplate`](../../api/grid/#exportdetailtemplate) event. In this event, format the detail row of the exported PDF document based on its parent row details.
+
+In the following sample, the detail template value has been structured with rows and cells format by specifying the [`columnCount`](../../api/grid/detailTemplateProperties/#columncount), [`columnHeader`](../../api/grid/detailTemplateProperties/#columnheader), and [`rows`](../../api/grid/detailTemplateProperties/#rows) parameters using its [`parentRow`](../../api/grid/exportDetailTemplateEventArgs/#parentrow) details. This allows you to create a PDF grid inside the detail row. Additionally, apply custom styles to specific cells using the [`style`](../../api/grid/detailTemplateCell/#style) property.
+
+> If a value for the `columnCount` is not provided, the PDF grid's columns in the detail row will be generated based on the count of the `columnHeader`/`rows` first row's `cells`(../../api/grid/detailTemplateRow/#cells).s
```
Comment (MaithiliyK) [inline ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md:26]: content formatting is not correct

File: ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+{% include code-snippet/grid/detail-template-export-cs1/index.html %}
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "page.domainurl/code-snippet/grid/detail-template-export-cs1" %}
+{% endif %}
+
+## Limitations
+
+* The export feature for detail templates is not supported with server-side exporting.
```
Comment (MaithiliyK) [inline ej2-javascript/grid/pdf-export/exporting-detail-template-grid.md:58]: move the limitation under server exporting

File: ej2-javascript/grid/pdf-export/exporting-with-templates.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+{% include code-snippet/grid/detail-template-export-cs1/index.html %}
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "page.domainurl/code-snippet/grid/detail-template-export-cs1" %}
+{% endif %}
+
+## Exporting with caption template
+
+PDF export allows exporting Grid with caption template to a PDF document.
```
Comment (florenceliliana) [inline ej2-javascript/grid/pdf-export/exporting-with-templates.md:97]: The PDF export feature allows for the exporting of a Grid with a caption template to a PDF document.

File: ej2-javascript/grid/pdf-export/exporting-with-templates.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+In the below sample, the customized caption text is exported to PDF by using [captionText](../../api/grid/exportGroupCaptionEventArgs/#captiontext) property in the [exportGroupCaption](../../api/grid/#exportgroupcaption) event.
```
Comment (florenceliliana) [inline ej2-javascript/grid/pdf-export/exporting-with-templates.md:99]: In the following sample, the customized caption text is exported to PDF using the [captionText](../../api/grid/exportGroupCaptionEventArgs/#captiontext) property in the [exportGroupCaption](../../api/grid/#exportgroupcaption) event.

**Review by florenceliliana — CHANGES_REQUESTED** [review]
The content review is completed.

**Review by MaithiliyK — APPROVED** [review]
changes are fine

**Review by jmfarook7 — APPROVED** [review]
fine

**Pull request merged**